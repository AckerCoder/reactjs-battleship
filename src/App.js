import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import "./App.css";
import Game from "./pages/game/game.component";
import HomePage from "./pages/homepage/homepage.component";
import ScoreTable from "./pages/score-table/score-table.component";
import SettingsPage from "./pages/settings-page/settings-page.component";
import YouLost from "./pages/you-lost/you-lost.component";

function App() {
  return (
    <div className="App">
      <Router>
        <Switch>
          <Route exact path="/">
            <HomePage />
          </Route>
          <Route path="/settings-page">
            <SettingsPage />
          </Route>
          <Route path="/game/:numberOfTurns/:playerName">
            <Game />
          </Route>
          <Route path="/score-table">
            <ScoreTable />
          </Route>
          <Route path="/you-lost">
            <YouLost />
          </Route>
        </Switch>
      </Router>
    </div>
  );
}

export default App;
