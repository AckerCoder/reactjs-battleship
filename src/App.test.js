import React from "react";
import { shallow } from "enzyme";
import App from "./app.js";

it("Expect to Match Snapshot", () => {
  expect(shallow(<App />)).toMatchSnapshot();
});

it("Expect to render App Component", () => {
  expect(shallow(<App />).length).toEqual(1);
});
