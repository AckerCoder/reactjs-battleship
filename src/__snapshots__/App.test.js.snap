// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to Match Snapshot 1`] = `
<div
  className="App"
>
  <BrowserRouter>
    <Switch>
      <Route
        exact={true}
        path="/"
      >
        <HomePage />
      </Route>
      <Route
        path="/settings-page"
      >
        <SettingsPage />
      </Route>
      <Route
        path="/game/:numberOfTurns/:playerName"
      >
        <Game />
      </Route>
      <Route
        path="/score-table"
      >
        <ScoreTable />
      </Route>
      <Route
        path="/you-lost"
      >
        <YouLost />
      </Route>
    </Switch>
  </BrowserRouter>
</div>
`;
